terror_error.go

```diff
@@ -284,10 +306,41 @@ func (e *Error) UnmarshalJSON(data []byte) error {
 	e.line = err.Line
 	e.file = err.File
 	e.message = err.Error
-	e.class = &ErrClass{
-		Description: className,
-		ID:          err.Class,
-		registry:    &Registry{Name: regName},
-	}
 	return nil
 }
+
+func (e *Error) Wrap(err error) *Error {
+	e.cause = err
```

Comment:
I think we should return a new instance of when `Wrap` a message instead of in-place changing the original instance. The error will be a global instance in many cases.